src/contracts/solver/SolverBase.sol

```diff
@@ -38,27 +40,31 @@ contract SolverBase {
         payBids(bids)
         returns (bool success, bytes memory data)
     {
+        // stored to allow solvers to call donateToBundler on correct execution environment
+        _executionEnv = msg.sender;
```

Comment:
Why is this necessary?  Searchers/Solvers could add this in if needed, but is there something blocking us from trusting msg.sender in this example?

Reply:
The `msg.sender` in `atlasSolverCall` (from SolverBase) is the Execution Environment, but then the solver calls itself using the solverOpData which targets the custom function selector the solver specified, calling their custom logic in the solver contract that inherited from SolverBase. At that point, the `msg.sender` becomes the solver contract, so I thought storing the original `msg.sender` in SolverBase would be a convenient way to give the solver access to the Execution Environment address.

Reply:
Ah! Good point!

I imagine most solvers will skip the "this.call{}()" step and just go straight to their func.  But in this case, I wonder if, for the SolverBase example, we might be better off appending the "msg.sender" to the calldata and using the classic meta tx pattern with a _msgSender() pure view func to read it.